lib/matplotlib/tests/test_axes.py

```diff
@@ -2267,6 +2267,115 @@ def test_grouped_bar_return_value():
         assert bc not in ax.containers
 
 
+def test_grouped_bar_single_hatch_str():
+    """All bars should share the same hatch when a single string is passed."""
+    fig, ax = plt.subplots()
+    x = np.arange(3)
+    heights = [np.array([1, 2, 3]), np.array([2, 1, 2])]
+    containers = ax.grouped_bar(heights, positions=x, hatch='//')
+
+    # Verify each bar has the same hatch pattern
+    for c in containers.bar_containers:
+        for rect in c:
+            assert rect.get_hatch() == '//'
+
+
+def test_grouped_bar_hatch_sequence():
+    """Each dataset should receive its own hatch pattern when a sequence is passed."""
+    fig, ax = plt.subplots()
+    x = np.arange(2)
+    heights = [np.array([1, 2]), np.array([2, 3]), np.array([3, 4])]
+    hatches = ['//', 'xx', '..']
+    containers = ax.grouped_bar(heights, positions=x, hatch=hatches)
+
+    # Verify each dataset gets the corresponding hatch
+    for gi, c in enumerate(containers.bar_containers):
+        for rect in c:
+            assert rect.get_hatch() == hatches[gi]
```

Comment:
slightly preferrable spelling:

```suggestion
    for hatch, c in zip(hatches, containers.bar_containers):
        for rect in c:
            assert rect.get_hatch() == hatch
```
Also applies to the other tests.